### ISSUE TITLE
Add a .raw_data attribute to FITS HDUs

### ISSUE THREAD

**Issue by embray** [astropy/astropy#4113]
As @leejjoon suggested way back in #993, it might be nice to have a convenient way to get at the raw, unscaled data of FITS arrays that have scale factors.

I always intended that to be a feature in some kind of FITS data class (that would also be used for columns of tables, so that individual columns can be accessed as scaled or raw).  

But in the meantime I think it would be fairly straightforward to add a `.raw_data` attribute to all HDU classes.  I think for the most part this could be implemented simply by wrapping the same code used by `.data`, but with `self._do_not_scale_image_data = True` set temporarily.  I just set "Effort-medium" because there may be some subtleties I haven't thought of yet, and there may also need to be some refactoring to avoid code duplication.

**Comment by embray** [comment]
Incidentally, I'm so excited for [`__numpy_ufunc__`](http://docs.scipy.org/doc/numpy-dev/reference/arrays.classes.html#numpy.class.__numpy_ufunc__).  Finally it will be more or less straightforward to make Numpy functions include automatic scaling of FITS data.  The `FITSData` class will be built around this, maybe with some less efficient fallback for Numpy < 1.10.

**Comment by mhvk** [comment]
Good idea. Note that, sadly, `__numpy_ufunc__` has been pushed to 1.11, as there was disagreement on exactly how `ndarray` methods should work with it.  It is top of the list for 1.11, though.

As for implementation, one might also consider having a `raw` attribute which returns a copy of self with scaling turned off, i.e., one would access the data via `hdu.raw.data`.

**Comment by embray** [comment]
D'oh--I remember hearing that it might be pushed back but I wasn't sure if that was the final word or not.

I don't think the indirection of `hdu.raw.data` is totally necessary.  If anything I'd prefer the other way around--`hdu.data.raw`. But then I'd have to go ahead and create a special class to do that.  Maybe not such a bad idea anyways.

**Comment by embray** [comment]
On the other hand, `hdu.raw` might not be a bad idea either.  I was already thinking of doing something similar to simplified the `CompImageHDU`, where the HDU returned to the user is a "logical" HDU that appears no different from an ImageHDU, while the underlying raw HDU would be the underlying `BinTableHDU` used to represent the compressed data (maybe with some additional options for tweaking the compression settings).

I guess a scaled image HDU could still be viewed as a "logical" HDU by default (as it is, when you access the data on a scaled ImageHDU it strips BSCALE and BZERO out of the header entirely, so it's as though they were never there at all.  This has caused no end of confusion, but is also a reasonable compromise, unfortunately.

**Comment by mhvk** [comment]
I considered suggesting `hdu.data.raw`, but thought it would be too much of a hassle, since now `data` cannot be a simple `ndarray` any more.  In that sense, the abstraction in your second comment of a "logical" HDU may end up being more intuitive.

**Comment by embray** [comment]
> since now data cannot be a simple ndarray any more

The plan is to do away with that anyways, since FITS data is not best represented by a simple ndarray.
